### ISSUE TITLE
Utilize collection environment (link declaration engine to collection environment inside of type checking)

### ISSUE THREAD

**Issue by emilyaherbert** [FuelLabs/sway#1824]
Once #1692 and #1693 go in, then we will be able to link the collection environment with the declaration engine to actually "turn on" this new compiler path.

Blocked by:
- [ ] #1692
- [ ] #1693

**Comment by emilyaherbert** [comment]
Closing in favor of #1692
